tools/importer/importer.py

```diff
@@ -0,0 +1,252 @@
+                        (' '.join(command)), returncode)
+            sys.exit(1)
+    return returncode, output
+
+def get_curr_sha(repo_path):
+    """ Gets the latest SHA for the specified repo
+    Args:
+    repo_path - path to the repository
+
+    Returns:
+    sha - last commit SHA
+    """
+    cwd = os.getcwd()
+    sha = None
+    os.chdir(abspath(repo_path))
+
+    cmd = "git log --pretty=format:%h -n 1"
+    ret, sha = run_cmd_with_output(cmd, exit_on_failure=True)
+
+    os.chdir(cwd)
+    return sha
+
+def check_branch(name):
+    """ Check if branch already exists in mbed-os local repository.
+    It will not verify if branch is present in remote repository.
+    Args:
+    name - branch name
+    Returns:
+    True - If branch is already present
+    """
+    branches = []
+    cmd = "git branch"
+    _, output = run_cmd_with_output(cmd, exit_on_failure=False)
+    if name in output:
+        return True
+    return False
+
+def branch_checkout(name):
+    """
+    Checkout the required branch
+    Args:
+    name - branch name
+    """
+    cmd = ['git', 'checkout', name]
+    run_cmd_with_output(cmd, exit_on_failure=False)
+
+def get_last_cherry_pick_sha(branch):
+    """
+    SHA of last cherry pick commit is returned. SHA should be added to all
+    cherry-pick commits with -x option.
+
+    Args:
+    branch - Hash to be verified.
+    Returns - SHA if found, else None
+    """
+    cmd = ['git', 'checkout', branch]
+    run_cmd_with_output(cmd, exit_on_failure=False)
+
+    SHA = None
+    get_commit = "git log -n 1"
+    _, output = run_cmd_with_output(get_commit, exit_on_failure=True)
+    lines = output.split('\n')
+    for line in lines:
+        if 'cherry picked from' in line:
+            SHA = line.split(' ')[-1]
+            SHA = SHA[:-1]
+    return SHA
+    #for words in output.split('\n') if 'origin' in line and not '->' in line]
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-l', '--log-level',
+                        help="Level for providing logging output",
+                        default='INFO')
+    parser.add_argument('-r',   '--repo-path',
+                        help="Git Repository to be imported",
+                        default=None)
+    parser.add_argument('-c', '--config-file',
+                        help="Configuration file",
+                        default=None)
+    args = parser.parse_args()
+
+    default = getattr(logging, 'INFO')
+    level = getattr(logging, args.log_level.upper(), default)
+
+    # Set logging level
+    logging.basicConfig(level=level)
+    rel_log = logging.getLogger("Importer")
+
+    if (args.repo_path is None) or (args.config_file is None) :
+        rel_log.error("Repository path and config file required as input. Use \"--help\" for more info.")
```

Comment:
If these arguments are required, could you mark them as required in the argument parser? `add_argument` takes a `required=True` parameter.